spec/lib/submission_builder/state_return_spec.rb

```diff
@@ -109,6 +109,96 @@
     end
   end
 
+  describe "Maryland W2 Box 14 STPICKUP handling" do
+    let(:builder_class) { SubmissionBuilder::Ty2024::States::Md::MdReturnXml }
+    let(:intake) { create(:state_file_md_intake, :df_data_2_w2s, filing_status: 'single') }
+    let(:submission) { create(:efile_submission, data_source: intake) }
+    let(:w2_node_name) { "IRSW2" }
+
+    context "when W2 has box14_stpickup" do
+      let!(:state_file_w2) do
+        create(
+          :state_file_w2,
+          state_file_intake: intake,
+          w2_index: 0,
+          box14_stpickup: 750.50,
+        )
+      end
+      before do
+        xml = Nokogiri::XML(intake.raw_direct_file_data)
+        intake.update(raw_direct_file_data: xml.to_xml)
+      end
+
+      it "includes STPICKUP in OtherDeductionsBenefitsGrp with rounded amount" do
+        xml = Nokogiri::XML::Document.parse(builder_class.build(submission).document.to_xml)
+        w2_node = xml.css(w2_node_name).first
+        expect(xml.css("OtherDeductionsBenefitsGrp Desc").map(&:text)).to include("STPICKUP")
+        expect(w2_node.css("OtherDeductionsBenefitsGrp Desc")[1]&.text).to eq "STPICKUP"
+        expect(w2_node.css("OtherDeductionsBenefitsGrp Amt")[1]&.text).to eq "751"
+      end
+
+      it "preserves existing OtherDeductionsBenefitsGrp entries" do
+        xml = Nokogiri::XML::Document.parse(builder_class.build(submission).document.to_xml)
+        w2_node = xml.css(w2_node_name).first
+
+        expect(w2_node.css("OtherDeductionsBenefitsGrp Desc")[0]&.text).to eq "414HSUB"
+        expect(w2_node.css("OtherDeductionsBenefitsGrp Amt")[0]&.text).to eq "250"
+      end
```

Comment:
since the set up is the same for both `it` blocks in this context, can we just put them all in the same `it` block & add comments if necessary?